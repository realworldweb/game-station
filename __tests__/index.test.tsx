import { render, screen } from '@testing-library/react';
import Home from '@/pages/index';

describe('Home', () => {
	it('renders homepage unchanged', () => {
		const { container } = render(<Home />);
		expect(container).toMatchSnapshot();
	});

	it('renders a heading', () => {
		render(<Home />);

		const heading = screen.getByRole('heading', {
			name: /Welcome to the game station/i,
		});

		expect(heading).toBeInTheDocument();
	});

	it('renders a link for naughts and crosses', () => {
		render(<Home />);

		const heading = screen.getByRole('link', {
			name: /Naughts and crosses/i,
		});

		expect(heading).toBeInTheDocument();
	});

	it('should be a link that have href value to "/naughts-and-crosses"', () => {
		render(<Home />);
		const link = screen.getByRole('link', { name: /Naughts and crosses/i });
		expect(link.getAttribute('href')).toBe('/naughts-and-crosses');
	});
});
