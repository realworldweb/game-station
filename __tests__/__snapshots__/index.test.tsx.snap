// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <h1
    class="px-3 py-5 text-white"
  >
    Welcome to the game station
  </h1>
  <div
    class="flex flex-col content-center justify-center gap-4 md:flex-row wrap"
  >
    <a
      class="relative rounded-md shadow-xl overflow-hidden mx-auto flex"
      href="/naughts-and-crosses"
      style="width: 310px; height: 220px;"
    >
      <img
        alt="game image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fnaughts-and-crosses.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fnaughts-and-crosses.webp&w=640&q=75 640w, /_next/image?url=%2Fnaughts-and-crosses.webp&w=750&q=75 750w, /_next/image?url=%2Fnaughts-and-crosses.webp&w=828&q=75 828w, /_next/image?url=%2Fnaughts-and-crosses.webp&w=1080&q=75 1080w, /_next/image?url=%2Fnaughts-and-crosses.webp&w=1200&q=75 1200w, /_next/image?url=%2Fnaughts-and-crosses.webp&w=1920&q=75 1920w, /_next/image?url=%2Fnaughts-and-crosses.webp&w=2048&q=75 2048w, /_next/image?url=%2Fnaughts-and-crosses.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <p
        class="p-0 px-3 text-sm absolute h-8 flex flex-wrap content-center justify-center bottom-0 bg-slate-900/80 rounded-tr-md text-white z-20"
      >
        Naughts and crosses
      </p>
    </a>
    <a
      class="relative rounded-md shadow-xl overflow-hidden mx-auto flex"
      href="/connect-four"
      style="width: 310px; height: 220px;"
    >
      <img
        alt="game image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fconnect-four.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fconnect-four.webp&w=640&q=75 640w, /_next/image?url=%2Fconnect-four.webp&w=750&q=75 750w, /_next/image?url=%2Fconnect-four.webp&w=828&q=75 828w, /_next/image?url=%2Fconnect-four.webp&w=1080&q=75 1080w, /_next/image?url=%2Fconnect-four.webp&w=1200&q=75 1200w, /_next/image?url=%2Fconnect-four.webp&w=1920&q=75 1920w, /_next/image?url=%2Fconnect-four.webp&w=2048&q=75 2048w, /_next/image?url=%2Fconnect-four.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <p
        class="p-0 px-3 text-sm absolute h-8 flex flex-wrap content-center justify-center bottom-0 bg-slate-900/80 rounded-tr-md text-white z-20"
      >
        Connect four
      </p>
    </a>
  </div>
</div>
`;
